golang/cosmos/x/swingset/keeper/keeper.go

```diff
@@ -76,6 +78,61 @@ func (k Keeper) SetParams(ctx sdk.Context, params types.Params) {
 	k.paramSpace.SetParamSet(ctx, &params)
 }
 
+func (k Keeper) GetBeansPerUnit(ctx sdk.Context) map[string]sdk.Uint {
+	params := k.GetParams(ctx)
+	beansPerUnit := make(map[string]sdk.Uint, len(params.BeansPerUnit))
+	for _, bpu := range params.BeansPerUnit {
+		beansPerUnit[bpu.Key] = bpu.Beans
+	}
+	return beansPerUnit
+}
+
+func (k Keeper) GetBeansOwing(ctx sdk.Context, addr sdk.AccAddress) sdk.Uint {
+	path := "beansOwing." + addr.String()
+	value := k.GetStorage(ctx, path)
+	if value == "" {
+		return sdk.ZeroUint()
+	}
+	return sdk.NewUintFromString(value)
+}
+
+func (k Keeper) SetBeansOwing(ctx sdk.Context, addr sdk.AccAddress, beans sdk.Uint) {
+	path := "beansOwing." + addr.String()
+	k.SetStorage(ctx, path, beans.String())
+}
+
+func (k Keeper) ChargeBeans(ctx sdk.Context, addr sdk.AccAddress, beans sdk.Uint) error {
+	beansPerUnit := k.GetBeansPerUnit(ctx)
+
+	owing := k.GetBeansOwing(ctx, addr)
+	owing = owing.Add(beans)
+
+	// Find the minimum debit.
+	beansPerMinFeeDebit := beansPerUnit[types.BeansPerMinFeeDebit]
+	beansToDebit := owing.Quo(beansPerMinFeeDebit)
```

Comment:
Do you need to multiply by `beansPerMinFeeDebit`? It looks like the intent here is to quantize the beans charged, e.g. if we charging multiples to 10 (`beansPerMinFeeDebit == 10`), and `owing = 23`, I think we'd want to set `beansToDebit` to 20 and `nowOwing` to 3, but this sets `beansToDebit` to 2.

If that's not what is intended, then please add a better comment and/or variable naming to adjust my expectations.

Reply:
Good catch!  I've rewritten this to be a simple subtraction of the remainder from the wasOwing.